internal/storage/storageutil/custom_retry_test.go

```diff
@@ -48,37 +41,60 @@ func (t customRetryTest) ShouldRetryReturnsTrueWithGoogleApiError() {
 		Body: "API rate limit exceeded",
 	}
 
-	ExpectEq(true, ShouldRetry(&err401))
-	ExpectEq(true, ShouldRetry(&err502))
-	ExpectEq(true, ShouldRetry(&err429))
+	assert.Equal(t, true, ShouldRetry(&err401))
+	assert.Equal(t, true, ShouldRetry(&err502))
+	assert.Equal(t, true, ShouldRetry(&err429))
 }
 
-func (t customRetryTest) ShouldRetryReturnsFalseWithGoogleApiError400() {
+func TestShouldRetryReturnsFalseWithGoogleApiError400(t *testing.T) {
 	// 400 - bad request
 	var err400 = googleapi.Error{
 		Code: 400,
 	}
 
-	ExpectEq(false, ShouldRetry(&err400))
+	assert.Equal(t, false, ShouldRetry(&err400))
 }
 
-func (t customRetryTest) ShouldRetryReturnsTrueWithUnexpectedEOFError() {
-	ExpectEq(true, ShouldRetry(io.ErrUnexpectedEOF))
+func TestShouldRetryReturnsTrueWithUnexpectedEOFError(t *testing.T) {
+	assert.Equal(t, true, ShouldRetry(io.ErrUnexpectedEOF))
 }
 
-func (t customRetryTest) ShouldRetryReturnsTrueWithNetworkError() {
-	ExpectEq(true, ShouldRetry(&net.OpError{
-		Err: errors.New("use of closed network connection")}))
+func TestShouldRetryReturnsTrueWithNetworkError(t *testing.T) {
+	assert.Equal(t, true, ShouldRetry(net.ErrClosed))
 }
 
-func (t customRetryTest) ShouldRetryReturnsTrueURLError() {
-	var urlErrConnRefused = url.Error{
-		Err: errors.New("connection refused"),
-	}
-	var urlErrConnReset = url.Error{
-		Err: errors.New("connection reset"),
+func TestShouldRetryReturnsTrueForConnectionRefusedAndResetErrors(t *testing.T) {
+	testCases := []struct {
+		name           string
+		err            error
+		expectedResult bool
+	}{
+		{
+			name:           "URL Error - Connection Refused",
+			err:            &url.Error{Err: errors.New("connection refused")},
+			expectedResult: true,
+		},
+		{
+			name:           "URL Error - Connection Reset",
+			err:            &url.Error{Err: errors.New("connection reset")},
+			expectedResult: true,
+		},
+		{
+			name:           "Op Error - Connection Refused",
+			err:            &net.OpError{Err: errors.New("connection refused")},
```

Comment:
let's write a case with error message in capitals. e.g. "Connection Refused"

Reply:
The tests would fail in that case. In general, error strings are not capitalised as per the golang style guide: [Ref](https://go.dev/wiki/CodeReviewComments#:~:text=Error%20strings%20should%20not%20be,usually%20printed%20following%20other%20context.)

Reply:
in that let's expand the error message by say "connection refused by peer"